### PR TITLE
[DO NOT MERGE] Trim user scopes before using in any operations

### PR DESCRIPTION
**Pull request by pkanher617**
This PR is to solve the issue raised in #845 where leading or trailing spaces are not trimmed from scope strings.

In this PR, I have changed the scope creation to trim before sending the scopes to be used in URL creation, and also trim them before sending them to the validation function. I have added test cases for this use case.

### CODE REVIEW

**Comment by jasonnutter** [comment]
@pkanher617 Something isn't right with coverage on this branch...

**Comment by pkanher617** [comment]
> @pkanher617 Something isn't right with coverage on this branch...

I had a .only in the tests, I have updated again.

File: lib/msal-core/test/utils/RequestUtils.spec.ts [hunk 1]
```diff
@@ -43,6 +43,21 @@ describe("RequestUtils.ts class", () => {
         expect(emptyScopesError.stack).to.include("RequestUtils.spec.ts");
     });
 
+    it.only("Scopes with leading or trailing spaces are trimmed before validation", () => {
```
Comment (jasonnutter) [inline lib/msal-core/test/utils/RequestUtils.spec.ts]: Ah, this is why coverage dipped.

Comment (jasonnutter) [reply lib/msal-core/test/utils/RequestUtils.spec.ts]: I'll add a lint rule to prevent `it.only`

File: lib/msal-core/src/utils/RequestUtils.ts [hunk 2]
```diff
@@ -46,6 +46,7 @@ export class RequestUtils {
         if(request) {
             // if extraScopesToConsent is passed in loginCall, append them to the login request; Validate and filter scopes (the validate function will throw if validation fails)
             scopes = isLoginCall ? ScopeSet.appendScopes(request.scopes, request.extraScopesToConsent) : request.scopes;
+            scopes = scopes ? scopes.map(scope => scope.trim()) : scopes;
```
Comment (jasonnutter) [inline lib/msal-core/src/utils/RequestUtils.ts]: Could this use the helper function you defined above?

Comment (pkanher617) [reply lib/msal-core/src/utils/RequestUtils.ts]: The helper is in ServerRequestParameters, not in RequestUtils

Comment (jasonnutter) [reply lib/msal-core/src/utils/RequestUtils.ts]: Right, could it go in a place where both can use it?

Comment (jasonnutter) [reply lib/msal-core/src/utils/RequestUtils.ts]: e.g. in the `ScopeSet` class

Comment (pkanher617) [reply lib/msal-core/src/utils/RequestUtils.ts]: Made some updates, take a look.

File: lib/msal-core/src/ScopeSet.ts [hunk 3]
```diff
@@ -41,19 +41,25 @@ export class ScopeSet {
...
+    static convertScopesToLowerCase(scopes: Array<string>): Array<string> {
         return scopes.map(scope => scope.toLowerCase());
     }
 
+    /**
+     * Trim scope values
+     * @param scopes 
+     */
+    static trimScopes(scopes: Array<string>): Array<string> {
+        return scopes ? scopes.map(scope => scope.trim()) : scopes;
```
Comment (jasonnutter) [inline lib/msal-core/src/ScopeSet.ts]: Could be `scopes && scopes.map(...)`.

Comment (pkanher617) [reply lib/msal-core/src/ScopeSet.ts]: Done.

**Pull request closed without merging**